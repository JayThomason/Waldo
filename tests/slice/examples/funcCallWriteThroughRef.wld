/** 
  * Note that because numList is passed from onCreate to otherFunc, it
  * gets written to through reference.  Ensures that numList shows up
  * in definite and conditional global/shared reads and writes for
  * onCreate.  Similarly, nothing definite should show up in otherFunc.
  * 
*/


PingPong

Endpoint Ping;
Endpoint Pong;


Sequences
{
}



Shared
{
}


/**** Ping endpoint *********/
Ping
{
    List (element: Number) numList = [1,3,4,];  
    onCreate ()
    {
        otherFunc(numList);
    }

    Public Function otherFunc(List (element: Number) someList)
    {
        someList[1] = 1;
    }
}


/**** Pong endpoint *********/
Pong
{
}
