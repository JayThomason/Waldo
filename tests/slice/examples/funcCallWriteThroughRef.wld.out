

{
    "conditionalGlobalSharedReads": [
        {
            "argPosition": -1, 
            "id": 0, 
            "mutable": 1, 
            "varName": "numList", 
            "varType": 1
        }
    ], 
    "conditionalGlobalSharedWrites": [
        {
            "argPosition": -1, 
            "id": 0, 
            "mutable": 1, 
            "varName": "numList", 
            "varType": 1
        }
    ], 
    "definiteGlobalSharedReads": [
        {
            "argPosition": -1, 
            "id": 0, 
            "mutable": 1, 
            "varName": "numList", 
            "varType": 1
        }
    ], 
    "definiteGlobalSharedWrites": [
        {
            "argPosition": -1, 
            "id": 0, 
            "mutable": 1, 
            "varName": "numList", 
            "varType": 1
        }
    ], 
    "funcName": "onCreate", 
    "readSet": [
        "numList"
    ]
}

{
    "conditionalGlobalSharedReads": [
        {
            "argPosition": 0, 
            "id": 9, 
            "mutable": 1, 
            "varName": "someList", 
            "varType": 3
        }
    ], 
    "conditionalGlobalSharedWrites": [
        {
            "argPosition": 0, 
            "id": 10, 
            "mutable": 1, 
            "varName": "someList", 
            "varType": 3
        }
    ], 
    "definiteGlobalSharedReads": [], 
    "definiteGlobalSharedWrites": [], 
    "funcName": "otherFunc", 
    "readSet": [
        "someList"
    ]
}





